libnavui-androidauto/src/main/java/com/mapbox/androidauto/notification/CarNotificationInterceptor.kt

```diff
@@ -0,0 +1,295 @@
+package com.mapbox.androidauto.notification
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.Matrix
+import android.graphics.drawable.Drawable
+import android.text.SpannableString
+import android.text.SpannableStringBuilder
+import android.text.TextUtils
+import android.text.format.DateFormat
+import androidx.car.app.model.CarColor
+import androidx.car.app.notification.CarAppExtender
+import androidx.car.app.notification.CarPendingIntent
+import androidx.core.app.NotificationCompat
+import androidx.core.content.ContextCompat
+import com.mapbox.api.directions.v5.models.BannerInstructions
+import com.mapbox.api.directions.v5.models.ManeuverModifier
+import com.mapbox.api.directions.v5.models.StepManeuver
+import com.mapbox.examples.androidauto.R
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.TimeFormat
+import com.mapbox.navigation.base.formatter.DistanceFormatter
+import com.mapbox.navigation.base.internal.maneuver.TurnIconHelper
+import com.mapbox.navigation.base.internal.time.TimeFormatter
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.formatter.MapboxDistanceFormatter
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.trip.session.NavigationSessionState
+import com.mapbox.navigation.core.trip.session.NavigationSessionStateObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.trip.notification.internal.NotificationTurnIconResources
+import java.util.Calendar
+
+/**
+ * Register this observer using [MapboxNavigationApp.registerObserver]. As long as it is
+ * registered, the trip notification will be shown in the car.
+ */
+@OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
+class CarNotificationInterceptor @JvmOverloads constructor(
```

Comment:
Will you add tests for this?

For example, the Bitmap created could be a renderer with instrumentation tests that verify it is rendering images as expected (for example [CarManeuverIconRendererTest](https://github.com/mapbox/mapbox-navigation-android/blob/7f67fc39f28e4abd752e7bd474910c33e655bb9c/libnavui-androidauto/src/androidTest/java/com/mapbox/androidauto/car/navigation/maneuver/CarManeuverIconRendererTest.kt#L38))

Reply:
Added. I didn't test the rendering though, because we don't render the notification itself, we just call `NotificationCompat.Builder` methods.